src/pages/Popup/Popup.tsx

```diff
@@ -778,25 +774,32 @@ const transitionToNextStateElement = (
 
 const Popup = () => {
   const [state, setState] = useBrowserStorageState(
-    POPUP_STATE_STORAGE_KEYS,
+    'popupState',
     PopupState.SignedOut
   );
   const [authStateSynced, setAuthStateSynced] = useState(false);
-  const [clientState, setClientState] = useState<ClientState>([]);
+  const [[setupUrl, webservices], setClientState] = useBrowserStorageState(
+    'clientState',
+    []
+  );
 
   useEffect(() => {
     const syncAuthState = async () => {
```

Comment:
Better do this

```ts
const { isLoading } = useBrowserStorageState('clientState')

const syncAuthState = async () => {
  if (isLoading) {
		// don't do anything if clientState is not loaded
    return
  }
	// do
}
```

Otherwise, it'll auth with the default URL first (because setupUrl is undefined). So everytime you open the Popup, it shows this screen first, even though you've authenticated with Chinese iCloud. Then it'll finally show the manage emails page. 

<img width="300" alt="Screenshot 2024-02-05 at 09 23 55" src="https://github.com/dedoussis/icloud-hide-my-email-browser-extension/assets/377544/ffcb999b-5070-42f4-a136-323637ada180">